homeassistant/components/sensor/yr.py

```diff
@@ -63,32 +69,25 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
 
     coordinates = dict(lat=latitude, lon=longitude, msl=elevation)
 
-    weather = YrData(coordinates)
-
     dev = []
     for sensor_type in config[CONF_MONITORED_CONDITIONS]:
-        dev.append(YrSensor(sensor_type, weather))
+        dev.append(YrSensor(sensor_type))
 
-    # add symbol as default sensor
-    if len(dev) == 0:
-        dev.append(YrSensor("symbol", weather))
-    add_devices(dev)
+    weather = YrData(hass, coordinates, dev)
+    tasks = [async_add_devices(dev), weather.async_update()]
+    yield from asyncio.gather(*tasks, loop=hass.loop)
```

Comment:
For now I will yield in two statements (1) to make sure entities are created and then (2) updated